CorsixTH/Src/main.cpp

```diff
@@ -52,6 +65,77 @@ inline void preload_lua_package(lua_State* L, const char* name,
       fn);
 }
 
+std::string search_script_file(lua_State* L) {
+  // 1. Check for --interpreter
+  int iNArgs = lua_gettop(L);
+  for (int i = 1; i <= iNArgs; ++i) {
+    if (lua_type(L, i) == LUA_TSTRING) {
+      size_t iLen;
+      const char* sCmd = lua_tolstring(L, i, &iLen);
+      if (iLen > 14 && std::memcmp(sCmd, "--interpreter=", 14) == 0)
+        return sCmd + 14;
+    }
+  }
+
+#ifdef CORSIX_TH_SEARCH_LOCAL_DATADIRS
+  // 2. Find CorsixTH.lua in working dir and program dir
+  static constexpr std::array<const char*, 4> asSearchDirs{
+      "./",
+      "CorsixTH/",
+      "Contents/Resources/",
+      "../Resources/",
+  };
+  std::string strProgramDir = "";
+  {
+    int iProgramPathLength = wai_getExecutablePath(nullptr, 0, nullptr);
+    if (iProgramPathLength != 0) {
+      char* sProgramDir = new char[iProgramPathLength + 1];
+      int iProgramDirLength;
+      int iProgramPathLengthReal = wai_getExecutablePath(
+          sProgramDir, iProgramPathLength + 1, &iProgramDirLength);
```

Comment:
whereami's documentation doesn't put a + 1 here. It might be safe because we don't use sProgramDir before injecting a null after iProgramDirLength, but there shouldn't be any reason to do it either.